### DISCUSSION TITLE
How to avid 'Route [filament.admin.auth.login] not defined.' ?

### DISCUSSION THREAD

**Discussion by magic-thomas** [filamentphp/filament#9229] [Help]
Hi. 

When I use filament, I don't create filament user ( which are created by 'php artisan make:filament-user' ). 

I just allow user to access a panel which is described in 'https://filamentphp.com/docs/3.x/panels/installation#allowing-users-to-access-a-panel' . 

But for guest user, 'Route [filament.admin.auth.login] not defined' error display for '/admin' url. 

How can I ignore this error and set it redirect to login or custom page ?

**Comment by leandrocfe** [comment]
Do you have `$panel->login()` configured?

**Reply by AzadGh95** [reply]
Correct!
add `$panel->login()` to **AdminPanelProvider.php**

example:
``` 
   return $panel
            ->id('admin')
            ->path('admin')
            ->login()
            .....
```

**Comment by ltroya-as** [comment]
Hi @leandrocfe, I am getting the same exact error on a fresh laravel project setting the `->path()` to `/`.

Not sure if it's because I am using corcel to allow wordpress users to authenticate on the filament admin.

Part of my `AdminPanelProvider.php`

```
return $panel
            ->default()
            ->id('admin')
            ->path('/')
            ->login()
```

If I use `->path(/admin)` it works as expected without any issue.

Also, if I run `php artisan route:list | grep login` I only see the following:

<img width="2994" alt="image" src="https://github.com/filamentphp/filament/assets/84338290/a77775c3-c071-4e52-a10d-d9085d722d78">

Here are the `filament` routes on the `route:list` command:

<img width="3004" alt="image" src="https://github.com/filamentphp/filament/assets/84338290/8334f156-9d3c-430f-b2fd-5091594ddbe5">

**Reply by ltroya-as** [reply]
Also, here is my `auth.php` config to configure [Corcel](https://github.com/corcel/corcel) 

```
'providers' => [
        'users' => [
            'driver' => 'corcel',
            'model' => env('AUTH_MODEL', App\Models\User::class),
        ],

        // 'users' => [
        //     'driver' => 'database',
        //     'table' => 'users',
        // ],
    ],
```

`App\Models\User.php`

```
<?php

namespace App\Models;

use Filament\Models\Contracts\FilamentUser;
use Filament\Models\Contracts\HasName;
use Filament\Panel;
use \Corcel\Model\User as CorcelUser;
use Illuminate\Database\Eloquent\Relations\HasMany;

class User extends CorcelUser implements FilamentUser, HasName
{
    protected static $aliases = [
        'id' => 'ID',
        'login' => 'user_login',
        'email' => 'user_email',
        'slug' => 'user_nicename',
        'url' => 'user_url',
        'nickname' => ['meta' => 'nickname'],
        'first_name' => ['meta' => 'first_name'],
        'last_name' => ['meta' => 'last_name'],
        'description' => ['meta' => 'description'],
        'created_at' => 'user_registered',
    ];

    protected $appends = [
        'id',
        'login',
        'email',
        'slug',
        'url',
        'nickname',
        'first_name',
        'last_name',
        'avatar',
        'created_at',
    ];

    public function getFilamentName(): string
    {
        return $this->email;
    }

    public function canAccessPanel(Panel $panel): bool
    {
        return true;
    }

    public function userCapabilities(): HasMany
    {
        return $this->hasMany(UserCapability::class, 'user_id');
    }
}
```

**Reply by ltroya-as** [reply]
Looks like it just happens even on a clean laravel 11 installation with Breeze. Here are the configurations I picked:

<img width="1016" alt="image" src="https://github.com/filamentphp/filament/assets/84338290/866b0cf1-971d-4c3c-b77c-03c89403f1e1">

**Reply by ltroya-as** [reply]
I found the issue, I just commented the route with the `->name('login')` on the `routes/auth.php` and everything works as expected.

On `No starter kit` projects, there is no issue at all.

**Comment by SanderMuller** [comment]
You can also try `php artisan filament:assets`

**Comment by yajuve** [comment]
I had the similar error with this message:
`Route [filament.admin.auth.profile] not defined.`
The original of my error is due that i had a custom **Profile** Page
`app/Filament/Pages/Profile.php`
Just delete it. In case someone faces the same issue.

**Reply by lqze** [reply]
> I had the similar error with this message: `Route [filament.admin.auth.profile] not defined.` The original of my error is due that i had a custom **Profile** Page `app/Filament/Pages/Profile.php` Just delete it. In case someone faces the same issue.

Wow, fun little bug!
